files-csharp/FastestWayToReadATextFileInCsharp/FastestWayToReadATextFileInCsharp/WaysToReadATextFileInCsharp.cs

```diff
@@ -1,86 +1,128 @@
+    
+    public string UseFileReadLines()
+    {
+        var lines = File.ReadLines(_sampleFilePath);
+        var stringBuilder = new StringBuilder();
 
-        public string UseFileReadAllText() => File.ReadAllText(_sampleFilePath);
-        
-        public string UseFileReadLines()
+        foreach(var line in lines)
         {
-            var stringArray = File.ReadLines(_sampleFilePath);
-
-            return string.Join(Environment.NewLine, stringArray);
+            stringBuilder.AppendLine(line);
         }
+        stringBuilder.Length -= Environment.NewLine.Length;
 
-        public string UseStreamReaderReadLine()
-        {
-            using var streamReader = new StreamReader(_sampleFilePath);
-            var stringBuilder = new StringBuilder();
-            string fileLine;
+        return stringBuilder.ToString();
+    }
 
-            while ((fileLine = streamReader.ReadLine()) != null)
-            {
-                stringBuilder.AppendLine(fileLine);
-            }
+    public string UseStreamReaderReadLine()
+    {
+        using var streamReader = new StreamReader(_sampleFilePath);
+        var stringBuilder = new StringBuilder();
 
-            return stringBuilder.ToString().TrimEnd();
+        while (streamReader.ReadLine() is { } fileLine)
+        {
+            stringBuilder.AppendLine(fileLine);
         }
+        stringBuilder.Length -= Environment.NewLine.Length;
 
-        public string UseStreamReaderReadToEnd()
-        {
-            using var streamReader = new StreamReader(_sampleFilePath);
+        return stringBuilder.ToString();
+    }
 
-            return streamReader.ReadToEnd();
-        }
+    public string UseStreamReaderReadToEnd()
+    {
+        using var streamReader = new StreamReader(_sampleFilePath);
 
-        public string UseStreamReaderReadBlock()
+        return streamReader.ReadToEnd();
+    }
+
+    public string UseStreamReaderReadBlock()
+    {
+        using var streamReader = new StreamReader(_sampleFilePath);
+        var buffer = new char[4096];
+        int numberRead;
+        var stringBuilder = new StringBuilder();
+
+        while ((numberRead = streamReader.ReadBlock(buffer, 0, buffer.Length)) > 0)
         {
-            using var streamReader = new StreamReader(_sampleFilePath);
-            var buffer = new char[1024];
-            int numberRead;
-            var stringBuilder = new StringBuilder();
+            stringBuilder.Append(buffer, 0, numberRead);
+        }
 
-            while ((numberRead = streamReader.ReadBlock(buffer, 0, buffer.Length)) > 0)
-            {
-                stringBuilder.Append(buffer, 0, numberRead);
-            }
+        return stringBuilder.ToString();
+    }
 
-            return stringBuilder.ToString();
+    public string UseStreamReaderReadBlockWithSpan()
+    {
+        using var streamReader = new StreamReader(_sampleFilePath);
+        var buffer = new char[4096].AsSpan();
```

Comment:
I should have mentioned this last time, but both here and the method above where we are allocating the buffer, we can gain some potential performance improvement, and certainly memory improvement by switching to an array from the Array pool.

Reply:
@jshergal From documentation : 

> Using the [ArrayPool<T>](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1?view=net-7.0) class to rent and return buffers (using the [Rent](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.rent?view=net-7.0) and [Return](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1.return?view=net-7.0) methods) can improve performance in situations where arrays are created and destroyed frequently, resulting in significant memory pressure on the garbage collector.

I think here, we are not allocating and destorying frequently the array, do you think it is a good usage for the ArrayPool?

Reply:
@karimkod - My suggestion was in light of the benchmark and in light of the potential usage of such a block of code. Since what we are talking about in this article is the most "performant" reading of text files, we are already dealing with the use case where we are trying to squeeze out some performance related to the reading of text files. If we want to keep what we already have and add a method that uses the ArrayPool, then we can showcase what performance gain (if any) there is, but more importantly can show the memory overhead improvement. Remeber in the case of a benchmark we are actually creating and allocating arrays frequently due to the nature of a benchmark which is executing this code multiple times to compute an average time and average memory usage.
So, that was what I had in mind with the suggestion. No pressure on the implementing. I was just adding the thought as we are focusing on performance, and this is another way to potentially squeeze out a little bit more.

Reply:
@jshergal thanks for the insight. I see the point clearly now. 
@cjaynjoku could you please add this as another method of the benchmark. You can put it in the same section as the one that uses the array in the article (no need for an extra section). And include it in the benchmark section as well. 
And also, if you could highlight the memory usage as @jshergal suggested above. 
Thanks for both of you!